pkg/scans/checksrunner.go

```diff
@@ -156,15 +144,32 @@ func (c *ChecksRunner) CreateScanChecks(id string) error {
 		return err
 	}
 
+	// The checksCreated var should be never nil for the scans that are using
+	// async creation, in any case we check this in order to avoid panics.
+	checksCreated := 0
+	if scan.ChecksCreated != nil {
+		checksCreated = *scan.ChecksCreated
+	}
+
+	// Recheck if the scan was already finished by other worker since the last query.
+	if *scan.CheckCount == checksCreated {
```

Comment:
Yes, that's precisely the issue. que query condition **was** right, but when the worker processes the scan, this condition is no longer valid, because other worker did the work.

Reply:
Right now I see it.